api/lib/models/RSVP/RSVP-data-mapper-impl.ts

```diff
@@ -0,0 +1,222 @@
+import { IDbResult } from '../../services/database';
+import { GenericDataMapper } from '../../services/database/svc/generic-data-mapper';
+import { MysqlUow } from '../../services/database/svc/mysql-uow.service';
+import { IUowOpts } from '../../services/database/svc/uow.service';
+import { Logger } from '../../services/logging/logging';
+import { IActiveHackathonDataMapper } from '../hackathon/active-hackathon';
+import { IRSVPDataMapper } from './index';
+import { RSVP } from './RSVP';
+
+@Injectable()
+export class RSVPDataMapperImpl extends GenericDataMapper
+    implements IAclPerm, IRSVPDataMapper {
+
+  public COUNT: string = 'rsvp:count';
+  public CREATE: string = 'rsvp:create';
+  public DELETE: string = 'rsvp:delete';
+  public READ: string = 'rsvp:read';
+  public READ_ALL: string = 'rsvp:delete';
+  public UPDATE: string = 'rsvp:update';
+  public tableName: string = 'RSVP';
+
+  protected pkColumnName: string = 'uid';
+
+  constructor(
+    @Inject('IAcl') acl: IAcl,
+    @Inject('MysqlUow') protected readonly sql: MysqlUow,
+    @Inject('IActiveHackathonDataMapper') protected readonly activeHackathonDataMapper: IActiveHackathonDataMapper,
+    @Inject('BunyanLogger') protected readonly logger: Logger,
+  ) {
+    super(acl);
+    super.addRBAC(
+      [this.DELETE],
+      [AuthLevel.DIRECTOR],
+      undefined,
+      [AuthLevel[AuthLevel.TEAM_MEMBER]],
+    );
+    super.addRBAC(
+      [this.COUNT],
+      [AuthLevel.TEAM_MEMBER],
+      undefined,
+      [AuthLevel[AuthLevel.VOLUNTEER]],
+    );
+    super.addRBAC(
+      [this.READ_ALL],
+      [AuthLevel.VOLUNTEER],
+      undefined,
+      [AuthLevel[AuthLevel.PARTICIPANT]],
+    );
+    super.addRBAC(
+      [this.CREATE, this.READ, this.UPDATE],
+      [AuthLevel.PARTICIPANT],
+    );
+  }
+
+  public delete(id: UidType): Promise<IDbResult<void>> {
+    const query = squel.delete({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName)
+        .where(`${this.pkColumnName} = ?`, id)
+        .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query(query.text, query.values, { stream: false, cache: false }),
+    ).pipe(
+      map(() => ({ result: 'Success', data: undefined })),
+    ).toPromise();
+  }
+
+  public get(id: UidType, opts?: IUowOpts): Promise<IDbResult<RSVP>> {
+    let queryBuilder = squel.select({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder
+      .where(`${this.pkColumnName} = ?`, id);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<RSVP>(
+      query.text,
+      query.values,
+      { stream: false, cache: true },
+    ))
+      .pipe(
+        map((event: RSVP) => ({ result: 'Success', data: event })),
```

Comment:
This will end up `map((event: RSVP[]) => ({ result: 'Success', data: event[0] }))` bc of the changes made to mysqluow.service.ts in the base branch. Pls make this change even if the ts compiler yells at you